Objetivo 2 de @manujurado1 sobre el repositorio de @ignaciotitos [IV-22-23]

manujurado1 opened the pull request:
Sobre la estructura del repositorio
* [x] ¿He seguido las mejores prácticas en nombre de las clases y ficheros y
   disposición de los mismos?

Sobre el análisis del problema
* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

Sobre la planificación y la programación
* [x] ¿Los issues representan un problema, y no una tarea?
* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están asignados a una HU?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit? Los issues sobre los que estoy trabajando, ¿han
      sido asignados al primer milestone?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Son los milestones sobre los que estoy trabajando productos
   mínimamente viables? ¿O tengo que solicitar al product manager que
   cree milestones adicionales o precise de qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

manujurado1 committed:
Avanza la tarea #9, closes #10, closes #11 porque se ha creado el directorio internal donde se situarán los ficheros con las clases, siguiendo las mejores prácticas de GO

manujurado1 committed:
closes #12 porque se ha creado la clase moneda con los atributos discutidos

manujurado1 committed:
closes #14 porque se ha creado la clase Recomendacion con los atributos discutidos

manujurado1 committed:
closes #13 porque se ha creado la clase Prediccion con los atributos discutidos

manujurado1 committed:
closes #9 porque se ha creado el fichero iv.yaml indicando el lenguaje de programación y la entidad

manujurado1 commented:
Hola @ignaciotitos, necesito que asignes el milestone 0 a este PR para terminar de cumplir la lista de comprobación.
De la misma manera, me gustaría que revisaras el código y me comentases si hay algo con lo que no estés conforme.
Muchas gracias!

ignaciotitos commented:
Asignado!

ignaciotitos commented:
Ya está listo para entregarse? Hay que mencionar a JJ o algo?

manujurado1 commented:
Una vez que tu apruebes los cambios subiré el PR al repositorio y mencionaré al profesor.

ignaciotitos reviewed (approved):
👍🏽

manujurado1 committed:
closes #16 porque se han cambiado los atributos para resolver el problema de la entidad y los objetos valor

closes #16 porque se han cambiado los atributos para resolver el problema de la entidad y los objetos valor

Fix iv.yaml

manujurado1 commented:
He hecho los cambios pertinentes a lo hablado en el último issue, ¿cómo lo ves?

ignaciotitos commented:
¿Por qué hay en moneda un struct Información?

ignaciotitos commented:
Y no se si es necesario hacer en Predicción un struct dentro de otro struct. Un poco redundante creo yo.

manujurado1 commented:
> ¿Por qué hay en moneda un struct Información?

Eso lo cree ya que me comentaste esto en el issue relacionado: sí, yo creo que la clase moneda con eso bastaría. Lo que sí que habría que crear es una estructura, un struct o algo en los que guardar los valores del PIB, Importaciones, Exportaciones, inflación...

Y como no estaban claros los valores del struct lo dejé vacío, pero puedo meter los 4 que me dijiste, y que se añadan más en el futuro si fuese necesario

manujurado1 commented:
> Y no se si es necesario hacer en Predicción un struct dentro de otro struct. Un poco redundante creo yo.

Si, fallo mio, se me olvidó quitarlo

ignaciotitos commented:
> 

Ah okey okey, está bien. En todo caso información ya estaría contenido en Predicción entiendo.

ignaciotitos commented:
> > Y no se si es necesario hacer en Predicción un struct dentro de otro struct. Un poco redundante creo yo.
> 
> Si, fallo mio, se me olvidó quitarlo

Vale perfecto.

ignaciotitos commented:
> > ¿Por qué hay en moneda un struct Información?
> 
> Eso lo cree ya que me comentaste esto en el issue relacionado: sí, yo creo que la clase moneda con eso bastaría. Lo que sí que habría que crear es una estructura, un struct o algo en los que guardar los valores del PIB, Importaciones, Exportaciones, inflación...
> 
> Y como no estaban claros los valores del struct lo dejé vacío, pero puedo meter los 4 que me dijiste, y que se añadan más en el futuro si fuese necesario

Ah okey okey, está bien. En todo caso información ya estaría contenido en Predicción entiendo.

manujurado1 commented:
> > > ¿Por qué hay en moneda un struct Información?
> > 
> > 
> > Eso lo cree ya que me comentaste esto en el issue relacionado: sí, yo creo que la clase moneda con eso bastaría. Lo que sí que habría que crear es una estructura, un struct o algo en los que guardar los valores del PIB, Importaciones, Exportaciones, inflación...
> > Y como no estaban claros los valores del struct lo dejé vacío, pero puedo meter los 4 que me dijiste, y que se añadan más en el futuro si fuese necesario
> 
> Ah okey okey, está bien. En todo caso información ya estaría contenido en Predicción entiendo.

Vale, entonces lo elimino y así queda todo más limpio

manujurado1 commented:
Todo arreglado a mi parecer, ¿cómo lo ves?

ignaciotitos commented:
Yo creo que está bien. Se puede empezar con eso.

manujurado1 commented:
@JJ Listo para revisión!

JJ reviewed (changes requested):
Por favor, mirad (los dos) los errores frecuentes en este objetivo y seguid el procedimiento adecuado para modelizar el problema que se traduce de las historias de usuario elegidas para el milestone.

manujurado1 committed:
closes #14 issue inválido

manujurado1 committed:
closes #13 issue inválido

manujurado1 committed:
closes #12 issue inválido

manujurado1 committed:
closes #17 porque se ha creado una clase moneda que resuelve el problema de cómo sabrá el usuario el dinero que tendrá en el futuro mediante la estimación del valor y el % de confianza

manujurado1 commented:
@JJ  
Tras replantear el procedimiento de modelado, está listo para revisión.
Gracias.

JJ commented on internal/moneda.go:
```diff
@@ -0,0 +1,10 @@
+package internal
+
+type Moneda struct {
```
La modelización entre entidad y objeto valor implica llegar a un objeto inmutable que será un objeto valor. La creación, destrucción, o asociación de este objeto inmutable a ciertos cambios o evoluciones serán parte de la modelización, que aquí no habéis hecho.

manujurado1 replied:
Gracias por la corrección, pero me surge una duda sobre el objeto valor inmutable. En mi cabeza, lo lógico sería que este valor fuera la moneda, la cuál tiene un nombre y un valor. El problema es que en la realidad el valor de una moneda está en continuo cambio, por lo que dejaría de ser inmutable. De ahí mi problema para encontrar ese objeto inmutable.

JJ replied:
Entonces será solo un valor, que tendrá que ser manejado por una entidad.

JJ reviewed (commented):
Por favor, revisad los conceptos de diseño dirigido por dominio y los errores frecuentes del objetivo 2.

manujurado1 committed:
closes #17 porque se ha creado un modelo con una entidad (cartera) y un objeto valor (moneda) que permite al usuario conocer el dinero que tendrá en el futuro

manujurado1 commented:
Se ha replanteado la modelización, cambiando la entidad y añadiendo un objeto valor inmutable.
Listo para revisión @JJ 
Muchas gracias.

JJ commented on internal/cartera.go:
```diff
@@ -0,0 +1,8 @@
+
+type Cartera struct {
+	ListaMonedas        []*Moneda //Lista con todas las monedas que posee el usuariocd  en su propiedad
+	ValorTotal          float64   //Cálculo del valor actual de su cartera en USD
```
¿Esto es un atributo o un resultado de una función? ¿Por qué tiene que ser un atributo? ¿Qué historia de usuario pide eso?

JJ commented on internal/cartera.go:
```diff
@@ -0,0 +1,8 @@
+type Cartera struct {
+	ListaMonedas        []*Moneda //Lista con todas las monedas que posee el usuariocd  en su propiedad
+	ValorTotal          float64   //Cálculo del valor actual de su cartera en USD
+	ValorFuturo         float64   //Estimación del valor de su cartera en un futuro, medido en USD
```
Insisto, ¿por qué tiene que ser un atributo?

JJ reviewed (changes requested):
Tienes que intentar realmente hacer una modelización, fijarte en una historia de usuario en concreto, y diferenciar entre estructuras de datos y resultados de las funciones. No parece que la modelización haya ido más allá de agrupar en una estructura de datos conceptos mencionados en la historia de usuario.

manujurado1 committed:
revertir lo hecho #17 porque el issue no estaba bien planteado

manujurado1 committed:
closes #18 porque se ha creado un objeto valor Moneda que recoge en una estructura el nombre de la moneda, la cantidad que se posee y el valor de la misma

manujurado1 committed:
closes #19 porque se ha creado una entidad llamada Cartera que gestiona los objetos valor Moneda. Contiene un atributo en el que está la lista de todas las monedas que posee el cliente

manujurado1 committed:
closes #17 porque se ha añadido un atributo al modelo entidad para cumplir la HU guardando la información que le permita saber el dinero que tendrá en el futuro

manujurado1 commented:
Listo para revisión @JJ.
He intentado separar la HU en issues pequeños y hacer una modelización adecuada.
Muchas gracias!

JJ commented on internal/moneda.go:
```diff
@@ -0,0 +1,7 @@
+
+type Moneda struct {
+	Nombre   string
+	Cantidad float64 //Cantidad de unidades que posee de dicha moneda
```
No es un objeto valor si tienes que cambiar la cantidad que tiene. Por favor, tratad de pensar un poco como se modeliza el dominio en objetos valor y entidades. Los objetos valor siempre son inmutables.

JJ commented on internal/cartera.go:
```diff
@@ -0,0 +1,6 @@
+
+type Cartera struct {
+	ListaMonedas      []*Moneda          //Lista con todas las monedas que posee el usuario en su propiedad
+	PrediccionMonedas map[string]float32 //Diccionario dónde a cada moneda se le asocia un % (positivo o negativo) de la estimación de modificación del valor de esa moneda en el futuro
```
Por favor, consulta los errores frecuentes. Los resultados no tienen por qué ser parte de los atributos de una clase, que no tienen que incluir más que el estado del objeto.

manujurado1 replied:
Vale, si no he entendido mal estos dos comentarios, la idea sería eliminar la cantidad del modelo de Moneda ya que es un atributo mutable y eso gestionarlo desde la entidad, al igual que retirar el atributo de la predicción ya que es más un resultado que un atributo verdad?.
Muchas gracias por la ayuda!

JJ replied:
No, la idea es que 
1. Plantees todos los problemas que salen de #6 y/o #7 en un issue. Los issues son problemas.
2. Trates de ir dividiendo ese problema en diferentes apartados, hasta que lleges a una serie de objetos inmutables, creando problemas adicionales relacionados con el primero.
3. Una vez que llegues a esos objetos inmutables, los programes y vayas volviendo por los issues, resolviendo el resto de los problemas con objetos de más complejidad.

Para ello tienes que entender todos los aspectos de lo que desean los clientes, desentrañando cada una de los términos mencionados.

No se trata de que si habla de "cantidades" crees una variable que se llama "Cantidad". Se trata de que modelices el problema, para lo cual tienes que entenderlo.

JJ reviewed (changes requested):
Tanto en la modelización del dominio como en la propia implementación, tienes algunos problemas. Por favor, soluciónalos en la siguiente iteración.

manujurado1 committed:
deshacer #17 #18 y #19 debido a que son issues inválidos

manujurado1 committed:
closes #21 porque se ha añadido un atributo para indicar el estado del valor del dinero del usuario

manujurado1 committed:
closes #22 porque se ha añadido el atributo cantidad a la estructura 'Dinero' para tener en cuenta la cantidad de dinero que posee el usuario

manujurado1 committed:
closes #23 porque se ha añadido un atributo que indique la devaluación/revaloración de la moneda

manujurado1 commented:
Listo para revisión @JJ 
He planteado en issues todos los problemas que salen de ambas HU y luego he separado estos problemas en objetos inmutables.
He tenido también en cuenta lo comentado en #18 centrándome en el valor y la inflación y limitándome a cumplir las HU.
Muchas gracias!

JJ commented on internal/dinero.go:
```diff
@@ -0,0 +1,7 @@
+package internal
+
+type Dinero struct {
```
Esta línea se desarrolla para resolver #21. Sin embargo, #21 habla de "moneda" y este tipo de llama "Dinero". También habla de valor *actual*, luego habrá un componente temporal, ¿no? ¿Dónde está ese valor temporal en la estructura de datos?

manujurado1 replied:
Vale, lo tengo en cuenta para nombrar el tipo de forma correcta con sus atributos necesarios

JJ commented on internal/dinero.go:
```diff
@@ -0,0 +1,7 @@
+package internal
+
+type Dinero struct {
+	Valor     float64 //Indicador que refleja el valor actual
```
Estos tres campos, a diferencia del nombre del tipo, tratan de solucionar #23. Ese issue no está asignado a ninguna historia de usuario, luego es incorrecto. Pero es que tampoco tiene mucho que ver lo que pone aquí con lo que ahí dice.
Adicionalmente: no se ponen comentarios en línea, sólo como descripción de la clase. Y se deben seguir las mejores prácticas en Go.

manujurado1 replied:
Cierto, voy a modificar el issue porque se me pasó asignarlo a una HU y buscar el código correcto que resuelva ese issue

JJ commented on internal/dinero.go:
```diff
@@ -0,0 +1,7 @@
+type Dinero struct {
+	Valor     float64 //Indicador que refleja el valor actual
+	Cantidad  float32 //Cantidad de dinero que posee el usuario
+	Inflacion float64 //Parámetro que indica la inflación que tendrá el dinero
```
¿Entiendes el concepto de inflación? ¿Cómo se puede hablar de la inflación que *tendrá* el dinero?

manujurado1 replied:
Según he mirado, la mejor forma es a través de la oferta y la demanda monetaria, por lo que lo enfocaré así

JJ reviewed (changes requested):
Una vez más, el problema no es fácil de comprender y hay que investigar un poco aunque sea la wikipedia. Pero aún así, seguir los procedimientos que se han explicado repetidamente en clase es la única forma de encontrar una solución que sea satisfactoria para el cliente.

manujurado1 committed:
deshacer #21 #22 y #23 ya que estaban mal planteados

manujurado1 committed:
closes #21 porque se ha creado una estructura de datos moneda la cual refleja el valor actual de dicha moneda

manujurado1 committed:
closes #22 porque se ha añadido a la estructura moneda un atributo cantidad que nos permite conocer la cantidad de esa moneda que posee el usuario con el fin de poder calcular su dinero actual y futuro

manujurado1 committed:
closes #23 ya que se ha añadido a la estructura de datos de moneda los atributos de oferta y demanda, necesarios para calcular la inflación de la moneda y así predecir su valor futuro

manujurado1 commented:
Listo para revision! @JJ 
He tenido en cuenta sus correcciones y considero que vuelve a estar listo para revisión.
Muchas gracias!

JJ commented:
Por favor, @ignaciotitos, revisa.

ignaciotitos commented on internal/moneda.go:
```diff
@@ -0,0 +1,8 @@
+type Moneda struct {
+	ValorActual float64
+	Cantidad    float32
+	Oferta      float64
```
No entiendo muy bien la necesidad de tener aquí oferta y demanda. Que es oferta y demanda, en base a que, algo muy abstracto a mi parecer.

manujurado1 replied:
Esos dos atributos solucionan el issue #23, y he elegido esta forma de solucionarlo ya que al buscar el internet los factores por los que una moneda varía su precio encontré lo siguiente: 
![image](https://user-images.githubusercontent.com/74008375/208885836-fbcd2a6f-cb6a-4227-86d7-5c352a149591.png)

ignaciotitos replied:
La oferta y la demanda se refiere respecto a otra moneda según entiendo. Tu también al realizar esto entiendes eso o lo dices respecto solo a una moneda.

manujurado1 replied:
Yo lo que entiendo es que el usuario tiene dinero, y el dinero se tiene en monedas y esas monedas tienen un valor, el cual varía en función de la oferta y la demanda, de ahí que haya optado por esos atributos. Cualquier moneda entiendo que varía su precio en función de la oferta y la demanda.

JJ replied:
@manujurado1 resolver un problema implica entender el problema; en este hito hay que hacer un modelo del problema y crear código para que ese modelo se use en la lógica de negocio. Por eso, aunque no hay que programar la lógica de negocio, hay que tener una mínima comprensión de la misma para que el código sea aprovechable por @ignaciotitos . @manujurado1, según tú, ¿de dónde va a sacar @ignaciotitos la "oferta" y la "demanda", en qué se miden, qué va a hacer con ellas y qué es lo que va a obtener?
Te recuerdo en todo caso que **debes crear un issue donde plantee todos los problemas que surjan de la historia de usuario**. Si no lo haces, esto acaba convirtiéndose en lo que se está convirtiendo ahora mismo, en una discusión sobre qué significan los términos que no conduce a una evolución eficiente del código que satisfaga al usuario.

manujurado1 replied:
@JJ Entiendo lo que dice, pero por más que trato de intentar entender la lógica de negocio, el tema de predecir valores de monedas me parece algo super abstracto y fuera de mi conocimiento, por eso intento plantear soluciones sencillas.
El issue #23 es el que plantea esta problemática, por lo que entiendo que lo ideal sería tener esta conversación dentro del issue verdad? Y una vez que se haya llegado a una conclusión, plasmarlo en el código para que esté listo para que @ignaciotitos pueda aprovecharlo.

JJ replied:
Lo he repetido varias veces (y está en los errores frecuentes), pero lo vuelvo a repetir aquí. Un issue es un problema. No es una conversación entre dos personas. Para contestar a la conversación, ya lo puede hacer @ignaciotitos directamente y tarda lo mismo. Se trata de que *tú* veas, a partir de las historias de usuario, qué problemas se plantean. *Ese* es el contenido de este objetivo, no mirar cómo se hace un Struct en Go. La ingeniería es resolver problemas, y eso es exactamente lo que se os pide aquí: que analicéis una serie de deseos de un cliente, y creéis un modelo que permita avanzar hacia la resolución de esos problemes.

manujurado1 committed:
closes #23 porque se ha añadido a la estructura de moneda una estructura que recoge todos los atributos necesarios para calcular la variación de valor de una moneda

manujurado1 commented:
Listo para revisión @JJ @ignaciotitos 
He cambiado los terminos de oferta y demanda que eran abstractos por los obtenidos tras leer esta [guía](https://guiasjuridicas.wolterskluwer.es/Content/Documento.aspx?params=H4sIAAAAAAAEAMtMSbF1jTAAASMTM1NDtbLUouLM_DxbIwMDS0NDQ3OQQGZapUt-ckhlQaptWmJOcSoAPKZdyzUAAAA=WKE), los cuáles son necesarios para establecer el valor de las monedas.

ignaciotitos commented on internal/moneda.go:
```diff
@@ -0,0 +1,16 @@
+type Moneda struct {
+	ValorActual         float64
+	Cantidad            float32
+	DatosMercadoDivisas struct {
```
Pensándolo mejor no se si es la mejor opción dejar todos estos campos dentro de esta clase moneda, ya que los atributos cuando se creen no van a cambiar, pero moneda irá cambiando en el tiempo.

ignaciotitos replied:
Creo que se podrían guardar en otro sitio aparte.

manujurado1 replied:
Pero esos datos van a variar depende de la moneda con la que se esté trabajando, ya que cada moneda tendrá sus propios datos, entonces entiendo que es lógico que vaya ligado a la clase Moneda. Y no creo que sea ningún impedimento que estos datos no cambien dentro de la estructura de datos aunque haya otros que si lo puedan hacer.

ignaciotitos replied:
No, porque esos datos van respecto del país. Por ejemplo la moneda euro pero va asociada a un PIB de españa, alemania, etc...

manujurado1 replied:
Entiendo lo que dices, pero siguiendo tu ejemplo, un euro vale lo mismo en españa que en francia, por lo cual sus datos del mercado de divisas van a ser el mismo, esos datos van en función de la moneda.

ignaciotitos replied:
Muchos de los datos que has puesto son a nivel nacional, por ejemplo NivelRentaNacional, NivelInversionDomestico, etc.. son datos que van a estar asociados a un país.

ignaciotitos replied:
En todo caso si eso es como lo pones se deberían de tener en cuenta todos los países que participan en la UE que tienen euro para poder ver así el valor real de la moneda, lo que me parece bastante complicado.

manujurado1 replied:
> Muchos de los datos que has puesto son a nivel nacional, por ejemplo NivelRentaNacional, NivelInversionDomestico, etc.. son datos que van a estar asociados a un país.

Respecto a esto, aunque creo que para hacer la predicción servirían los datos de cualquier país, se me ocurre la opción de que una moneda tenga como atributo una lista de "DatosMercadoDivisas" ya que puede haber diferentes conjuntos de datos en función del país en caso de que la moneda se use en +1 país. Ves buena idea esto que te propongo?

manujurado1 replied:
Aunque siendo minimalistas, para cumplir la HU con tener los datos de un país ya es suficiente para estimar la inflación y así poder cumplir con el deseo del usuario.

ignaciotitos commented:
Está siendo un poco complicado de entender, para mí también. Vamos a hacerlo de la predicción no solo de una moneda sino de una en base a otra, en como actúa el mercado FOREX, por ejemplo el EUR / USD, por lo que para eso se necesitan una serie de datos, como el open, close, high, low.. te recomiendo que mires y te informes un poco más. Creo que así quedará un poco más claro.

JJ commented:
> Está siendo un poco complicado de entender, para mí también. Vamos a hacerlo de la predicción no solo de una moneda sino de una en base a otra, en como actúa el mercado FOREX, por ejemplo el EUR / USD, por lo que para eso se necesitan una serie de datos, como el open, close, high, low.. te recomiendo que mires y te informes un poco más. Creo que así quedará un poco más claro.

Te lo dije desde el principio. Tenéis que plantear un problema claro y del que tengáis conocimiento personal. Si decides cambiar cualquier cosa, recuerda que debe tener una lógica de negocio no trivial y cambiar los milestones e historias de usuario, así como el texto del objetivo 1, de acuerdo con esto.

ignaciotitos commented:
Creo que con lo que he dicho de FOREX se puede hacer un problema y sigue siendo el mismo planteamiento realmente.

ignaciotitos commented:
Lo único que cambia son los datos.

manujurado1 committed:
closes #23 porque se ha creado un objeto valor inmutable que recoje todas las características necesarias para poder hacer una predicción del valor de una moneda

manujurado1 commented:
Listo para revisiton! @JJ @ignaciotitos 
He cambiado la resolución del último issue para que siga el camino que quiere tomar @ignaciotitos para seguir con su proyecto.
Muchas gracias!

ignaciotitos reviewed (approved):
Perfecto! 👍🏽

ignaciotitos commented:
En vez de eso cambia los valores por los valores necesarios para calcular la inflación, así estará mejor.

manujurado1 committed:
closes #23 porque se ha añadido un atributo con una lista de indices de precios lo que nos permite calcular la inflación de la moneda

manujurado1 commented:
Tras hablar con @ignaciotitos sobre lo que habéis hablado esta mañana, está listo para revisión. @JJ

ignaciotitos reviewed (approved):
Yo creo que está bien.

JJ commented on internal/moneda.go:
```diff
@@ -0,0 +1,7 @@
+
+type Moneda struct {
+	ValorActual   float64
+	Cantidad      float32
```
¿Qué es exactamente "Moneda"? ¿Es un objeto valor? ¿El Indice de Precios es diferente para cada una de las cantidades? ¿Qué diferencia hay entre ValorActual y Cantidad?

JJ reviewed (changes requested):
Insisto que hacer un modelo del problema va más allá de ponerle nombres a las variables vagamente relacionados con el objeto de la aplicación. Consiste en diferenciar los diferentes tipos de datos, y hacere un análisis dirigido por el dominio para que se entienda perfectametne cómo se puede manejar desde la lógica de negocio. Para hacerlo, hay que formular una serie de problemas como issues, lo que no se ha hecho aquí.
El objeto valor que creas no es inmutable, y si lo es, ¿vas a repetir el índice de precios para cualquier cantidad diferente?
No se ha hecho ningún análisis sobre qué significa el "índice de precios", ni de qué forma indica cómo se va a devaluar o reevaluar. Por favor, consultad los errores frecuentes sobre el objetivo 2 y todo el material que se ha impartido al respecto, incluyendo los objetivos de aprendizaje de prácticamente todas las semanas desde finales de octubre.

manujurado1 commented:
>  Para hacerlo, hay que formular una serie de problemas como issues, lo que no se ha hecho aquí.

Los issues #20 #21 #22 #23 son problemas que se han encontrado para satisfacer la HU #7, ¿Están mal hechos esos issues? Y en el caso de que estén mal, que entiendo que si por lo la lína a la que estoy respondiendo, ¿qué tengo que cambiar en la forma de hacerlos para que sean válidos?
Muchas gracias!

manujurado1 committed:
Deshacer #21 #22 #23 porque estaban mal planteados

manujurado1 committed:
closes #25 porque se ha creado un objeto valor que soluciona el issue de como calcular el IPC

manujurado1 committed:
closes #24 porque se ha creado un objeto valor que soluciona la problemática de como calcular la inflación

manujurado1 committed:
closes #23 closes #20 porque se ha creado la entidad predicción que contiene los atributos necesarios para predecir el dinero que tendrá el usuario, ya que contiene la cantidad de dinero actual del usuario y las tasas de inflación que permite saber la variación de valor de la moneda

manujurado1 commented:
Listo para revisión! @JJ 
He cambiado el planteamiento, creando a mi parecer correctamente los issues como problemas y recorriendolos solucionandolos con objetos valor.
He usado [esta página](https://bbvaassetmanagement.com/es/como-se-mide-la-inflacion-el-ipc/#:~:text=El%20%C3%ADndice%20se%20calcula%20dividiendo,da%20como%20resultado%20el%20IPC.) de guía.
Muchas gracias!

JJ commented:
> > Para hacerlo, hay que formular una serie de problemas como issues, lo que no se ha hecho aquí.
> 
> Los issues #20 #21 #22 #23 son problemas que se han encontrado para satisfacer la HU #7, ¿Están mal hechos esos issues? Y en el caso de que estén mal, que entiendo que si por lo la lína a la que estoy respondiendo, ¿qué tengo que cambiar en la forma de hacerlos para que sean válidos? Muchas gracias!

Sí, si lo están. Si quieres vamos por issue.

- #20 Tiene sólo un commit c1c961931e65a844304c3cdf11d51411a29ef7d1 que crea un struct llamado `Prediccion` (¿de que?) que tiene una "CantidadDineroActual" (¿cuando?) y "TasasInflacion" (¿desde cuando? ¿cada cuanto? ¿Para quien? ¿Sólo para el usuario? ¿Hay diferentes tasas de inflación dependiendo del dinero? ¿Del usuario?) El mensaje de commit no sigue las mejores prácticas (50 caracteres en la primera línea), modifica un fichero que no tiene nada que ver con el mismo y además sólo describe lo que hace, y no de qué forma constituye una solución. La `Prediccion` incluye, aparentemente, `CantidadDineroActual`, lo que es algo sorprendente. Lo peor de todas formas es que la pregunta, formulada en este issue tiene una respuesta simple: una variable llamada "DineroActual", que, desgraciadamente, no ayuda en lada porque ni tiene que ver con el usuario (¿es que la devaluación de una cantidad *realmente* tiene algo que ver con un usuario?) ni en realidad es un modelo de nada. Y no es un modelo de nada, porque plantear preguntas *no es* plantear problemas, y más si introducen elementos que nada tienen que ver con la lógica de negocio (¿la lógica de negocio necesita "usuarios"?)
- #23 tiene una gran cantidad de commits, de los cuales el último es este mismo que hemos indicado antes c1c961931e65a844304c3cdf11d51411a29ef7d1. Menciona la "moneda" de la que no hay trazas en el commit correspondiente, que aparentemente trata de "Predicción".

Si quieres continúo, pero no creo que sirva de nada, porque ya has vuelto a enviar el código a revisión sin solucionar lo que se te había comentado y sin tratar de entender por qué los issues que plantean preguntas no son automáticamente problemas, si son irrelevantes y se pueden contestar de forma que sea irrelevante a la lógica de negocio. Te rogaría que, para optimizar nuestro tiempo en estas fiestas, te informaras correctamente de todos los errores comunes en este objetivo dos y de los objetivos de aprendizaje correspondiente, especialmente cómo hay que descomponer un problema usando domain driven design hasta llegar a objetos valor y eventualmente a entidades programables.

También rogaría a @ignaciotitos que continuara siendo de ayuda considerando el código enviado y viendo de qué forma puede servirle, o no, a la lógica de negocio que tiene pensada.

Finalmente, en estas revisiones se trata de ver si entendéis los conceptos subyacentes y sabéis ponerlos en práctica. 

> ¿qué tengo que cambiar en la forma de hacerlos para que sean válidos? Muchas gracias!

Ponerlos en práctica implica que *tú* cambies lo que consideres conveniente para que efectivamente sean problemas que resuelvas con el código escrito. El objetivo no es el código, es que aprendáis a usar una metodología eficiente para escribir código que resuelva problemas de clientes.

manujurado1 committed:
Deshacer cambios

closes #25 closes #24 closes #23 closes #20 issues inválidos

manujurado1 committed:
Añadir la estructura de datos referente al objeto valor TasaInflacion

closes #27 porque se ha creado la estructura de datos correspondiente al objeto valor TasaInflacion, con todas los atributos necesarios.

manujurado1 committed:
Creacion estructura de datos Prediccion que actúa como entiedad de los objetos valor

closes #26 porque se ha creado la estructura de datos referente a la prediccion que se desea hacer, que será la entidad que maneje los objetos valor de las tasas de inflacion.

manujurado1 commented:
Buenos días, antes que nada felices fiestas a ambos.
He vuelto a plantear los issues y generar una estructura que satisfaga la HU y sirva para la lógica de negocio.
@ignaciotitos coméntame cuando puedas si te parece correcto o si hay algo que no te cuadre, y cuando me des tu visto bueno, etiquetaré al profesor.
Muchas gracias!

ignaciotitos commented on internal/prediccion.go:
```diff
@@ -0,0 +1,7 @@
+
+type Prediccion struct {
+	Mes            int
+	Anio           int
```
Porque se repite el mes y el año en predicción cuando ya está en TasaInflacion?

manujurado1 replied:
Son diferentes, cada tasa de inflación tiene un mes y un año asociado ya que el valor de esa tasa de inflación es la variación del PIB en esa fecha.
Mientras tanto, la predicción tiene un mes y un año a predecir, es decir, la fecha sobre la que el usuario quiere saber la predicción.
¿Me he explicado bien?

ignaciotitos replied:
Es la variación del *IPC*. Sí, entiendo.

JJ replied:
> Son diferentes, cada tasa de inflación tiene un mes y un año asociado ya que el valor de esa tasa de inflación es la variación del PIB en esa fecha.

La variación del PIB no tiene nada que ver con la inflación. Puede haber crecimiento del PIB con aumento del IPC, disminución del mismo, o estancamiento. En todo caso como digo arriba, la predicción es el resultado de la lógica de negocio, no forma parte del dominio del problema que hay que modelizar en este milestone.

> Mientras tanto, la predicción tiene un mes y un año a predecir, es decir, la fecha sobre la que el usuario quiere saber la predicción. ¿Me he explicado bien?

En ese caso, no forma parte de este milestone.

manujurado1 replied:
Perdon, fue fallo mío al explicarle a mi compañero, que me lío un poco con las siglas, me refería al valor del IPC.

Y respecto a lo segundo que dice, lo puse por lo mismo que le he comentado antes, guiándome en la HU

ignaciotitos reviewed (approved):
👍🏽

manujurado1 commented:
Muchas gracias, etiqueto ahora al profesor para que lo revise.
Listo para revisión! @JJ

JJ commented:
Os agradecería que consultárais y siguiérais las buenas prácticas en mensajes de commits. La primera, la primera línea debe tener 50 caracteres y mencionar el issue en el que se esté trabajando. Esto hace más fácil su corrección, y también se ayuda a pensar siempre que lo principal es trabajar con el issue.

JJ commented on internal/prediccion.go:
```diff
@@ -0,0 +1,7 @@
+package internal
+
+type Prediccion struct {
```
La `Prediccion`, ¿es parte del problema? Si no lo es, sino que es donde habría que depositar el resultado de la lógica de negocio, ¿os parece que pertenece a este *milestone*?

manujurado1 replied:
Si el fin es complacer la HU #6, dónde dice que quiere saber su cantidad de dinero en el futuro, considero que la predicción es necesaria para saber el valor del dinero en el futuro.

JJ replied:
Si lo que estás diciendo es que predicción == cantidad de dinero que va a tener en el futuro, ¿no sería simplemente una cantidad entonces?
En todo caso, mira el [milestone actual](https://github.com/ignaciotitos/IV---HealthForAll/milestone/1) y lo que incluye. No parece que incluya nada que se llame "predicción". Y una vez más, consultar los errores frecuentes en este objetivo.

manujurado1 replied:
Si, pero esa cantidad sería el resultado de aplicar la lógica de negocio, y como es un resultado por eso no lo he puesto.
Y respecto a lo del milestone, lo he visto, pero en este caso considero que difiere un poco la HU asociada al mismo, por lo que me cuesta ver con claridad en que me tengo que enfocar realmente

JJ replied:
Creo que no estás entendiendo lo que estamos haciendo aquí y qué es lo que te estoy diciendo. No te estoy pidiendo que justifiques lo que has hecho. Te estoy diciendo que si no es parte del problema, no es parte de este milestone, y por tanto no te corresponde a ti hacerla, sino a @ignaciotitos cuando llegue a ese punto. También te estoy diciendo que modelizar resultados de lógica de negocio en una clase es erróneo, porque los resultados de funciones o métodos no tienen por qué modelizarse, son simplemente resultados. Finalmente, te estoy diciendo que no has llegado a esta solución con la metodología correcta, y que por tanto deberías comenzar a seguir una metodología correcta para llegar a un resultado aceptable.
Por lo tanto, tomo nota de lo que tú consideras y te vuelvo a repetir que sería conveniente que tuvieras en cuenta estas otras consideraciones para, finalmente, superar el objetivo.

JJ commented on internal/tasaInflacion.go:
```diff
@@ -0,0 +1,7 @@
+package internal
+
+type TasaInflacion struct {
```
¿Esta es la tasa anual? ¿Mensual?

manujurado1 replied:
Mensual ya que tiene como atributos un mes y un año, pero quizás podría haber puesto un nombre más descriptivo si

JJ replied:
> Mensual ya que tiene como atributos un mes y un año, pero quizás podría haber puesto un nombre más descriptivo si

Correcto.

JJ commented on internal/tasaInflacion.go:
```diff
@@ -0,0 +1,7 @@
+package internal
+
+type TasaInflacion struct {
+	Mes   int
```
Esto puede valer -3

manujurado1 replied:
No sabía que tenía que controlar yo las restricciones de valor en estos casos, lo tendré en cuenta para el siguiente intento

JJ replied:
Tienes que usar los tipos de datos que sean más adecuados para reflejar las restricciones que tendrías que haber incluido en la descripción del problema que este código resuelve. Si el lenguaje elegido no tiene estructuras de datos suficientemente expresivas para esto, habrá que hacerlo en el constructor de las mismas o creando estructuras de datos de mayor nivel (entidades u objetos valor) que lo hagan.

manujurado1 replied:
Vale, le veo sentido. Muchas gracias

JJ replied:
Venga, ánimo. Ya queda menos.

JJ commented on internal/tasaInflacion.go:
```diff
@@ -0,0 +1,7 @@
+
+type TasaInflacion struct {
+	Mes   int
+	Anio  int
```
Y esto puede valer 35000

JJ reviewed (changes requested):
Aunque al menos la definición de la tasa de inflación se acerca bastante más a estar correcto que antes
* La metodología para llegar a ella no es correcta; seguimos en el proceso Q-A que ya he repetido que es erróneo, en vez de plantear los diferentes problemas que se planteen.
* De esa forma, no se ha llegado a plantear una estructura de datos que sea correcta, porque no cumple las restricciones que tienen que tener los diferentes elementos ni tiene un nombre suficientemente descriptivo.
* Finalmente, se incluyen en este objetivo cosas que no pertenecen al *milestone* actual.

manujurado1 commented:
> Aunque al menos la definición de la tasa de inflación se acerca bastante más a estar correcto que antes
> 
> * La metodología para llegar a ella no es correcta; seguimos en el proceso Q-A que ya he repetido que es erróneo, en vez de plantear los diferentes problemas que se planteen.
> * De esa forma, no se ha llegado a plantear una estructura de datos que sea correcta, porque no cumple las restricciones que tienen que tener los diferentes elementos ni tiene un nombre suficientemente descriptivo.


Sobre la metodología, he hecho las preguntas que consideraba necesarias para clarificar la lógica de negocio y poder realizar una estructura de datos útil. Pero si no es la correcta, ¿hay algún documento o algo por el estilo en el que pueda aprender como realizarlo correctamente? Ya que quizás se me haya pasado mirarlo y por eso no termino de hacer bien las cosas.

JJ commented:
> Sobre la metodología, he hecho las preguntas que consideraba necesarias para clarificar la lógica de negocio y poder realizar una estructura de datos útil. Pero si no es la correcta, ¿hay algún documento o algo por el estilo en el que pueda aprender como realizarlo correctamente? Ya que quizás se me haya pasado mirarlo y por eso no termino de hacer bien las cosas.

Hay que empezar por hacer caso a las indicaciones que se hacen. Pera empezar, no se trata de hacer "una estructura de datos útil", sino *un modelo del problema*, como además dice en el *milestone* en el que estás trabajando. Adicionalmente, hay una lista de comprobación que está ahí para algo. En los [errores frecuentes de la semana 10](https://github.com/JJ/IV-22-23/blob/7e0bf0f1f8a3521074011f9ea3050da3f48f23d9/sesiones/semana-10.md?plain=1#L63) dice:

> Usar la lista de comprobación como tal, y no como "cosas que hay que marcar
  para pasar el objetivo*.

Los "errores frecuentes" son problemas en los que ha incurrido una o varias personas en la clase, y se recopilan principalmente para que no se vuelva a caer. Os he aconsejado repetidamente que lo hagáis.

Durante tres clases seguidas, además, llevamos a cabo un hackatón, donde tratamos de emular este mismo proceso de modelización de un problema y trabajo en equipo. Se podía participar tanto local como remotamente. Es conveniente ir a clase, porque en ella a veces se dicen cosas que pueden ayudar a sacar la asignatura adelante, y en todo caso permiten al profesor ayudar al estudiante personalmente a sacar las cosas.

Adicionalmente, hay documentación (enlazada desde las sesiones correspondientes, y a su vez enlazadas desde el grupo de Telegram de la asignatura) donde se explica qué es domain driven design (también presentado en clase varias veces, y bajo demanda cuando alguien lo ha solicitado) y otras buenas prácticas como código limpio. También se explica qué es una revisión de un PR, y cómo en ella, a través de las indicaciones, se trata de que el estudiante llegue a superar el objetivo.

> Sobre la planificación y la programación
> 
>     * [x]  ¿Los issues representan un problema, y no una tarea?

Esto lo has marcado tú. Si no entendías qué significaba, ¿por qué lo has marcado como hecho?

manujurado1 commented:
> > Sobre la metodología, he hecho las preguntas que consideraba necesarias para clarificar la lógica de negocio y poder realizar una estructura de datos útil. Pero si no es la correcta, ¿hay algún documento o algo por el estilo en el que pueda aprender como realizarlo correctamente? Ya que quizás se me haya pasado mirarlo y por eso no termino de hacer bien las cosas.
> 
> Hay que empezar por hacer caso a las indicaciones que se hacen. Pera empezar, no se trata de hacer "una estructura de datos útil", sino _un modelo del problema_, como además dice en el _milestone_ en el que estás trabajando. Adicionalmente, hay una lista de comprobación que está ahí para algo. En los [errores frecuentes de la semana 10](https://github.com/JJ/IV-22-23/blob/7e0bf0f1f8a3521074011f9ea3050da3f48f23d9/sesiones/semana-10.md?plain=1#L63) dice:
> 
> > Usar la lista de comprobación como tal, y no como "cosas que hay que marcar
> > para pasar el objetivo*.
> 
> Los "errores frecuentes" son problemas en los que ha incurrido una o varias personas en la clase, y se recopilan principalmente para que no se vuelva a caer. Os he aconsejado repetidamente que lo hagáis.
> 
> Durante tres clases seguidas, además, llevamos a cabo un hackatón, donde tratamos de emular este mismo proceso de modelización de un problema y trabajo en equipo. Se podía participar tanto local como remotamente. Es conveniente ir a clase, porque en ella a veces se dicen cosas que pueden ayudar a sacar la asignatura adelante, y en todo caso permiten al profesor ayudar al estudiante personalmente a sacar las cosas.
> 
Me hubiera gustado ir y sé que me hubiera ayudado a entenderlo todo mejor, pero por motivos de trabajo me ha sido imposible ir a ninguna clase, por eso he preguntado por documentación al respecto, para intentar entenderlo mejor.

> Adicionalmente, hay documentación (enlazada desde las sesiones correspondientes, y a su vez enlazadas desde el grupo de Telegram de la asignatura) donde se explica qué es domain driven design (también presentado en clase varias veces, y bajo demanda cuando alguien lo ha solicitado) y otras buenas prácticas como código limpio. También se explica qué es una revisión de un PR, y cómo en ella, a través de las indicaciones, se trata de que el estudiante llegue a superar el objetivo.
> 
Volveré entonces a mirar tanto el grupo y las sesiones para ver si consigo terminar de entenderlo. Gracias!
> > Sobre la planificación y la programación
> > ```
> > * [x]  ¿Los issues representan un problema, y no una tarea?
> > ```
> 
> Esto lo has marcado tú. Si no entendías qué significaba, ¿por qué lo has marcado como hecho?

Lo he marcado porque cuando he ido haciendo los intentos por superar el objetivo he hecho por cumplirlo

JJ commented:
> > Esto lo has marcado tú. Si no entendías qué significaba, ¿por qué lo has marcado como hecho?
> 
> Lo he marcado porque cuando he ido haciendo los intentos por superar el objetivo he hecho por cumplirlo

La lista de comprobación está para que el estudiante vea qué conceptos son necesarios y se interese por entenderlos y ponerlos en práctica. Puedes marcarlo o no marcarlo, eso no influye en la superación del objetivo. Pero si lo marcas, asegúrate que lo estás entendiendo, porque si no lo haces, es doble trabajo para la persona que revisa.

manujurado1 commented:
> Aunque al menos la definición de la tasa de inflación se acerca bastante más a estar correcto que antes
> 
> * La metodología para llegar a ella no es correcta; seguimos en el proceso Q-A que ya he repetido que es erróneo, en vez de plantear los diferentes problemas que se planteen.
> * De esa forma, no se ha llegado a plantear una estructura de datos que sea correcta, porque no cumple las restricciones que tienen que tener los diferentes elementos ni tiene un nombre suficientemente descriptivo.

He estado revisando toda la documentación sobre la metodología y como plantear los diferentes problemas y me ha surgido una duda: ¿Todos los problemas que se plantean se plantean desde el punto de vista del usuario? Porque en este caso no veo muchos problemas desde el punto de vista de lo que al usuario le llega, sino que veo los verdaderos problemas alrededor de lo necesario para poder aplicar la lógica de negocio. Y por esta razón no termino de ver claros los problemas que se generan y plantear los mismos como issues.

Muchas gracias por la ayuda!

JJ commented:
> > Aunque al menos la definición de la tasa de inflación se acerca bastante más a estar correcto que antes
> > 
> > * La metodología para llegar a ella no es correcta; seguimos en el proceso Q-A que ya he repetido que es erróneo, en vez de plantear los diferentes problemas que se planteen.
> > * De esa forma, no se ha llegado a plantear una estructura de datos que sea correcta, porque no cumple las restricciones que tienen que tener los diferentes elementos ni tiene un nombre suficientemente descriptivo.
> 
> He estado revisando toda la documentación sobre la metodología y como plantear los diferentes problemas y me ha surgido una duda: ¿Todos los problemas que se plantean se plantean desde el punto de vista del usuario? Porque en este caso no veo muchos problemas desde el punto de vista de lo que al usuario le llega, sino que veo los verdaderos problemas alrededor de lo necesario para poder aplicar la lógica de negocio. Y por esta razón no termino de ver claros los problemas que se generan y plantear los mismos como issues.

No. El punto de vista del usuario está reflejado en las historias de usuario. Los problemas/issues son siempre desde el punto de vista del implementador. ¿Cómo hago para satisfacer al usuario modelizando el dominio del problema correctamente? Al usuario no le interesa que uses un entero o un entero sin signo para los meses, o un enum que sea mucho más estricto en las comprobaciones. A *ti* te interesa que si usas un entero para el mes en el que se registra el IPC, sea mucho menos práctico que simplemente meter todos los datos de inflación en un array que vaya cambiando según va avanzando el tiempo. O que el dato de inflación se suele publicar con una sola cifra significativa, y por tanto si hay más de una debe haber algún error. Muchos de los problemas relativos a datos específicos son en realidad cómo incluirlos o representarlos pero también puede haber otros adicionales como relaciones entre ellos (por ejemplo, no puedes tener dos datos de inflación diferentes con el mismo mes y día). La *modelización* consiste en usar código para representar esto de la mejor forma posible, de forma que sea el tipo de dato el que impida que se incluya en él un estado erróneo y no haya que escribiri código para comprobarlo (y además, de forma continua)

manujurado1 commented:
> 



> No. El punto de vista del usuario está reflejado en las historias de usuario. Los problemas/issues son siempre desde el punto de vista del implementador. ¿Cómo hago para satisfacer al usuario modelizando el dominio del problema correctamente? Al usuario no le interesa que uses un entero o un entero sin signo para los meses, o un enum que sea mucho más estricto en las comprobaciones. A ti te interesa que si usas un entero para el mes en el que se registra el IPC, sea mucho menos práctico que simplemente meter todos los datos de inflación en un array que vaya cambiando según va avanzando el tiempo. O que el dato de inflación se suele publicar con una sola cifra significativa, y por tanto si hay más de una debe haber algún error. Muchos de los problemas relativos a datos específicos son en realidad cómo incluirlos o representarlos pero también puede haber otros adicionales como relaciones entre ellos (por ejemplo, no puedes tener dos datos de inflación diferentes con el mismo mes y día). La modelización consiste en usar código para representar esto de la mejor forma posible, de forma que sea el tipo de dato el que impida que se incluya en él un estado erróneo y no haya que escribiri código para comprobarlo (y además, de forma continua)

Vale, creo que lo he entendido todo, voy a ponerme a plantear los issues de nuevo y a ver si consigo hacerlo bien o al menos estar más cerca de conseguirlo.
Muchas gracias por la ayuda!

manujurado1 committed:
Deshacer cambios relacionados con issues inválidos

manujurado1 committed:
closes #29 se ha creado un objeto valor TasaInflacionMensual

Se ha creado un objeto valor que reune el valor de la tasa de inflación de un determinado mes junto a los atributos mes y año para identificar la fecha a la que se refiere

manujurado1 committed:
closes #30 porque se ha implementado un constructor que evita fechas inválidas

manujurado1 committed:
closes #28 creación Entidad manejadora de lista de tasas de inflacion

Se ha creado una entidad que maneja los objetos valor de las tasas de inflación ya que es el elemento acordado con el que se asocia la variación de valor de una moneda.

manujurado1 committed:
closes #31 porque ya no se permiten tasas de inflación con la misma fecha

Se ha creado una función para añadir tasas de inflación a la lista propia del atributo de la entidad que comprueba que ese par de valores (mes,año) no está ya añadido

manujurado1 commented:
Listo para revisión @JJ 

He revisado todo el material impartido de nuevo y corregido los errores intentando seguir las directrices que me ha ido comentando.
Espero esta vez estar mejor encaminado.
Muchas gracias!

JJ commented:
De veras, @manujurado1, intentad id un poco más allá de lo inmediato que os he corregido para no introducir nuevos errores. Prácticametne en todas las semanas desde sexta o séptima está entre los objetivos de aprendizaje o entre los errores frecuentes algo relacionado con los mensajes de commits y las buenas prácticas que hay que seguir. El hacerlo así facilita la vida del que corrige, pero también pone a la persona que lo escribe en el marco mental de estructurar la información que hay que introducir de una forma determinada. Entre esas convenciones, es que los mensajes de commit se estructuran en una primera línea de 50 caracteres, luego un espacio en blanco, y luego líneas de 80 caracteres (aunque en realidad da igual). Si no lo haces así, mira lo que pasa:
![Captura de pantalla de 2022-12-28 19-43-13](https://user-images.githubusercontent.com/500/209858231-10edb007-e9a5-4988-960c-4c427a59ea47.png)
Los mensajes se cortan, y no se puede ver lo que se ha escrito.
Entre esas buenas prácticas está también que no se repitan los metadatos que hay en un commit, que incluye lo que se ha creado o cambiado, los ficheros modificados. ¿Te parece que

> [creación Entidad manejadora de lista de tasas](https://github.com/ignaciotitos/IV---HealthForAll/pull/15/commits/ffd91677c259fab0e041ed9812d13420e8a3b8a1)

añada información sobre lo que es obvio mirando al commit? Un mensaje de commit debe servir para que quien trabaje con ese código, mirando "blame"

![Captura de pantalla de 2022-12-28 19-45-19](https://user-images.githubusercontent.com/500/209858467-e4e1f914-53ef-40d5-b30f-97274fe784da.png)

pueda tener una idea por qué se ha hecho eso, qué soluciona, y qué suposiciones tiene detrás. De este mensaje sólo sabemos que es una entidad, lo que parece más o menos obvio, pero no por qué es un array de `TasaInflacionMensual` y no simplemente de números reales (o quizás racionales, porque sólo se expresan con una cifra).

En ingeniería hay que aprender metodologías, porque las metodologías se pueden aplicar a la resolución de cualquier problema, en cualquier lenguaje. Os rogaría que os esforzárais por hacerlo, porque es la forma de crear aplicaciones de forma eficiente (y de superar este objetivo)

JJ reviewed (approved):
A pesar de los comentarios, creo que es válido y se puede usar directamente en la lógica de negocio. Seguir buenas prácticas desde el principio te habría traído hasta este punto mucho más rápidamente, sin embargo. Te agradecería que lo pusieras en práctica en el resto de los objetivos. Ánimo.

manujurado1 commented:
Perfecto, prestaré especial atención en los mensajes de commit de aquí en adelante para evitar esos errores y que sean lo más legibles y útiles posibles y me informaré bien de todas las buenas prácticas para que el resto de objetivos vayan más fluidos.
Muchas gracias! 👍
